### PR TITLE
[Feat] 노선도 홈 알림 안내 바·역 정보 패널 노선 표시·실시간/시간표 토글·방면 제목 UI 개선

### LINKED ISSUE #2200: [Feat] 노선도 홈 알림 안내 바·역 정보 패널 노선 표시·실시간/시간표 토글·방면 제목 UI 개선
## 목표

노선도 홈의 4개 UI — 상단 신규 알림 안내 바, 하단 역 정보 패널의 이전-현재-다음역 노선 표시, 실시간/시간표 토글, "○○ 방면" 제목 — 를 오너 확정 디자인 스펙(2026-07-16)대로 개선한다. 그 외 UI, 특히 방면 제목 아래 열차 정보 행("○○행"·도착 시간)은 무변경으로 보존한다.

## 배경

- 현재 상단 알림은 운행 공지 전용 풍선형 배너이고, 알림함 새 항목은 벨 버튼 배지로만 표시된다. 오너 결정에 따라 운행 공지 또는 알림함 새 항목 중 하나라도 있으면 상단바에 붙는 전체 너비 안내 바 하나로 통합하고, 탭하면 알림함으로 이동한다.
- 하단 역 정보 패널의 이전-현재-다음역 표시가 선택 노선색이 아닌 고정 강조색을 사용해 노선 정체성이 드러나지 않는다. 좌우 바와 중앙 캡슐 테두리·노선 배지를 모두 선택 노선색으로 통일한다.
- 실시간/시간표 전환이 라벨만 바뀌는 단일 버튼이라 현재 상태와 대안이 한눈에 보이지 않는다. 옆의 새로고침 아이콘은 실시간 데이터 새로고침 의도였으나 주변역 재탐색에 잘못 연결되어 있어 제거한다(실시간은 자동 갱신, 위치 재탐색은 GPS 버튼이 별도).
- 방면 제목이 단색이라 노선 맥락이 약하다. 역명은 선택 노선색, " 방면"은 무채색으로 분리한다.

### 실기기 QA 후속 (오너 확정 2026-07-17, 이 이슈에 포함)

- 실시간/시간표 토글의 선택 칸 외곽이 배경과 구분이 약해 시인성이 나쁘다 — 기존 박스형 토글이 쓰던 색 토큰을 외곽·글자에 적용한다.
- 노선도에서 역을 탭하면 팬 메뉴만 뜨고 하단 역 정보 패널이 열리지 않는다 — 검색 선택 플로우와 동일하게 역 탭 시 해당 역의 하단 패널을 함께 연다.
- GPS 현재 위치가 캐시(last-known) 전용이라 캐시가 없으면 "현재 위치를 확인하지 못했어요"로 실패한다 — 캐시 미스 시 능동 단발 위치 요청(타임아웃 포함)을 추가한다.

## 범위

- `apps/mobile/lib/main.dart`: 통합 알림 바 배선(기존 disruptionBanner 슬롯·알림함 이동 콜백 재사용)
- `apps/mobile/lib/features/notifications/presentation/`, `features/service_notice/presentation/`: 통합 알림 바 신설, 기존 풍선형 배너 홈 사용 제거
- `apps/mobile/lib/network_map.dart`: 패널 노선 표시(노선색 단일 소스)·2버튼 토글·방면 제목·구분선
- `apps/mobile/lib/accessible_design.dart`: 신규 색 토큰(알림 바 웜톤, 토글 비선택, 구분선)
- `apps/mobile/test/`: 신규 widget test, 시각 가드 ratchet 상한 조정(w900·w800·radius>8 — 오너 결정 사유 주석 박제)
- `apps/mobile/android/.../MainActivity.kt`: currentLocation 채널에 캐시 미스 시 능동 단발 위치 요청·타임아웃 추가 (QA 후속)

## 비범위

- "○○행"·도착 예정 시간·실시간/시간표 값 등 열차 정보 행의 위젯·폰트·색·간격·데이터 포맷
- 알림함 화면·운행 공지 목록 화면·알림 상태 로직(NoticeController 등)
- 상단바 높이·구성, 바텀시트 툴바 높이(52dp), 라우팅

## 완료 조건

- [ ] 새 알림(운행 공지 또는 알림함 항목)이 있을 때만 전체 너비 안내 바가 상단바 바로 아래 간격 0·radius 0·그림자 없이 표시되고, 없으면 빈 공간까지 사라진다. 바 또는 "알림 보기" 탭 시 알림함으로 정확히 1회 이동한다.
- [ ] 이전-현재-다음역 표시의 좌우 바·중앙 캡슐 테두리·노선 배지가 모두 선택 노선색이고(2호선이면 양측 #00A84D), 환승역에서 노선 변경 시 동시 갱신된다.
- [ ] 실시간/시간표가 2버튼 토글(각 59×48, 툴바 52dp 유지)로 즉시 전환되고 전환 애니메이션이 없으며 선택 상태가 접근성 서비스에 노출된다. 토글 옆 새로고침 버튼은 제거된다.
- [ ] 방면 제목이 역명(선택 노선색)+" 방면"(무채색) 한 줄로 표시되고, 아래 열차 정보 행 무변경이 회귀 테스트로 검증된다.
- [ ] 토글 선택 칸 외곽·글자가 기존 박스형 토글 색 토큰으로 표시되어 배경과 구분된다 (QA 후속).
- [ ] 노선도에서 역을 탭하면 팬 메뉴와 함께 그 역의 하단 역 정보 패널이 열리고 body(노선 바·방면·열차 정보)가 렌더된다. 해석 불가 역은 기존 동작(팬 메뉴만)을 유지한다 (QA 후속).
- [ ] GPS 탭 시 캐시된 위치가 없어도 능동 위치 요청으로 주변역 패널이 열리고, 타임아웃 시에만 기존 오류 문구를 표시한다 (QA 후속).
- [ ] 320/375/480dp 너비에서 overflow 없음. dart format·flutter analyze·flutter test 통과.
- [ ] 변경 전/후 스크린샷 증거, full PR 템플릿 본문, GitHub PR Review 증거 확인.

## 위험 및 완화

- #2181 presentation 이동 리팩터와 `main.dart`·features 경로가 겹친다 — 먼저 병합되는 쪽 기준으로 나중 쪽이 rebase한다.
- 시각 가드 ratchet 조정은 필요 최소 건수만 올리고 오너 결정 날짜와 이 이슈 번호를 주석으로 남긴다.
- 풍선형 공지 배너 제거로 운행 공지 상세 진입은 알림함·운행 공지 목록 경로로 유지된다.

### PR DESCRIPTION
**Pull request by AquilaXk**
## 관련 이슈

close #2200

## 작업 배경

- 상단 알림이 운행 공지 전용 풍선형 배너와 벨 버튼 배지로 나뉘어 있어, 오너 결정(2026-07-16)에 따라 새 알림이 있으면 상단바에 붙는 전체 너비 안내 바 하나로 통합한다.
- 하단 역 정보 패널의 이전-현재-다음역 표시가 선택 노선색이 아닌 고정 강조색이라 노선 정체성이 드러나지 않는다.
- 실시간/시간표 전환이 라벨만 바뀌는 단일 버튼이라 현재 상태와 대안이 한눈에 보이지 않고, 옆의 새로고침 아이콘은 실시간 데이터 새로고침 의도였으나 주변역 재탐색에 잘못 연결되어 있었다.
- 방면 제목이 단색이라 노선 맥락이 약하다.

## 작업 내용

- 통합 신규 알림 안내 바 `NewNotificationBar` 신설: 운행 공지 disruption 또는 알림함 새 항목이 있으면 상단바 바로 아래 간격 0·전체 너비·radius 0·그림자 없음·하단 테두리 1dp로 표시, 탭 시 알림함으로 정확히 1회 이동, 없으면 빈 공간 없이 제거. 기존 풍선형 운행 공지 배너 삭제 (`6515d9b2`)
- 이전-현재-다음역 노선 바 `NearbyStationLineBar`: 좌우 바·중앙 캡슐 테두리·원형 노선 배지를 모두 선택 노선색 단일 소스(카탈로그 색 → 폴백 매핑)로 통일(2호선 양측 #00A84D), 양끝 반원·중앙 흰 캡슐 Stack 오버레이, LayoutBuilder 비율(margin 3.476%·바 348:28·중앙 37.069%·캡슐 129:41) (`8265d8f9`)
- 실시간/시간표 2버튼 세그먼트 토글 `NearbyDataSourceToggle`(각 59×48, radius 16, 선택=흰 배경+브랜드 시그니처 2dp 테두리), 전환 애니메이션 없음, Semantics selected 노출, 잘못 배선된 툴바 새로고침 버튼 제거 (`8265d8f9`)
- "○○ 방면" 제목 `NearbyDirectionTitle`: 역명=선택 노선색·" 방면"=#2F2F2F, 14sp w900, 실시간·시간표 열 공유. 시간표는 행마다 반복되던 방면 라벨을 열 상단 헤더 1개로 이동(시각 표시는 무변경). 열차 정보 행("○○행"·도착 시간)은 위젯·폰트·색·간격·데이터 포맷 무변경 (`8265d8f9`)
- 신규 색은 전부 `accessible_design.dart` 토큰으로 추가, 시각 가드 ratchet(w900·w800·radius>8·soft 틴트 예외)을 오너 결정 필요 최소 건수로 조정하고 각 줄에 근거 주석 박제 (`434ee0a4`)

## 검증

- 실행한 명령과 결과:
  - `dart format --output=none --set-exit-if-changed lib test` → 0 changed (exit 0)
  - `flutter analyze` → No issues found!
  - `flutter test` 전체 → +1328 All tests passed! (design_guard·accessibility_baseline·widget_test·onboarding_app_flow 포함, origin/main a8e3212b rebase 후 재실행)
  - 신규 widget test 19건: 알림 바 표시/미표시/탭 1회/margin·radius 0, 2호선 양측 #00A84D, 노선 탭 변경 시 바·캡슐·배지 동시 갱신, 토글 상태 전환·애니메이션 부재·refresh 부재, 방면 제목 색 분리, 열차 정보 행 스타일 회귀, 320/375/480dp overflow 없음

## 검증 증거

| 항목 | 플랫폼 | 확인 방법 | 증거 | 결과 |
| --- | --- | --- | --- | --- |
| 알림 없음 시 바 미표시 | Android | 에뮬레이터(API 36) debug 설치 후 홈 확인 | 로컬 evidence: `.superpowers/evidence/2200/after-home.png` (오너 루트) | 바·빈 공간 없음 확인 |
| 알림 있음 바·탭 이동 | Android/iOS 공통 | widget test (`test/new_notification_bar_test.dart`) | flutter test 통과 로그 | 통과 |
| 노선색·토글·방면 제목 | Android/iOS 공통 | widget test (`test/network_map_nearby_panel_test.dart`) | flutter test 통과 로그 | 통과 |
| 패널·토글·방면 실기기 시각 확인 | Android | 오너 수동 QA (이 PR의 QA 단계) | 에뮬레이터 위치 주입으로 주변역 패널 진입 불가해 자동 캡처 미확보 — 오너 QA 인계 항목 | 대기 |
| 접근성(대비·터치 영역·라벨) | Android/iOS 공통 | `test/accessibility_baseline_test.dart` | flutter test 통과 로그 | 통과 |

## Version impact

- [x] no version change
- [ ] mobile patch
- [ ] mobile minor
- [ ] mobile major
- [ ] backend deploy only
- [ ] datapack release only
- [ ] route/realtime contract change
- [ ] DB migration change

## Route commercialization gate impact

- [x] route-commercialization-gate.json 영향 없음
- [ ] route ETA accuracy, realtime coverage, accessibility regression, route v2 contract report를 갱신했다.
- [x] 상용 경로/ETA claim을 추가하거나 변경하지 않는다.

## Route release readiness tracker impact

- [x] route-release-readiness-tracker.json 영향 없음
- [ ] #1414 하위 release blocker issue 또는 production evidence 완료 조건을 갱신했다.
- [x] 실시간/교통약자 길찾기 출시 준비 완료 claim을 추가하거나 변경하지 않는다.

### Version decision

- mobile versionName: 변경 없음
- mobile versionCode: 변경 없음
- datapack version: 변경 없음
- route contract: 변경 없음
- realtime contract: 변경 없음
- backend identity: 변경 없음

## 리뷰어 메모

- 리뷰어가 먼저 봐야 할 지점:
  - `network_map.dart`의 private 위젯을 테스트 가능하도록 public 위젯 4개(features/network_map·notifications presentation)로 추출 — 로직 동일성 여부
  - 시간표 열의 방면 라벨을 행 반복에서 열 헤더 1개로 hoist한 것이 "열차 정보 행 무변경" 경계와 부합하는지 (시각 텍스트·스타일은 무변경)
  - `design_guard_test.dart` ratchet 조정이 필요 최소 건수인지 (w900 1건·w800 이동·radius 신규 파일 2건, 그림자·그라데이션·splash 밴 무변경)
  - 툴바 새로고침 버튼 제거로 `onRetry` 배선이 남는 에러 상태 경로

## 리스크

- 주변역 패널 실기기 시각(캡슐 겹침 연속성·반원 끝·배지 축소 가독성)은 위젯 테스트로 픽셀 검증이 불가해 오너 QA에서 확인 필요
- 병행 리팩터(#2173 계열)와 같은 파일(`network_map.dart`, `features/home`)을 만져 후속 PR rebase 충돌 가능

## 체크리스트

- [x] PR 본문은 이 템플릿 섹션을 삭제하지 않고 모두 채웠다.
- [x] CI 결과를 확인했다.
- [x] CodeRabbit 리뷰를 확인했다.
- [x] GitHub PR Review 객체가 있는지 확인했다. CodeRabbit status check만으로는 리뷰 완료로 보지 않는다.
- [ ] CodeRabbit 실행이 불가능하거나 PR Review 객체가 없으면 Codex CLI code review를 단일 PR review로 게시했다.
- [x] 배포 영향이 있는 경우 CD 상태를 확인했다. (배포 영향 없음)

## Summary by CodeRabbit

* **새로운 기능**
  * 홈 화면에 통합 알림 안내 바를 추가했습니다.
  * 주변역 패널에서 실시간 도착 정보와 시간표를 전환할 수 있습니다.
  * 주변역 정보에 선택 노선 색상, 방향 제목, 도착 정보 및 역 연결 표시를 적용했습니다.
  * 주요 안내 요소에 접근성 라벨과 선택 상태 정보를 강화했습니다.

* **개선 사항**
  * 기존 운행 공지 배너를 새로운 알림 안내 바로 통합했습니다.
  * 다양한 화면 너비에서 주변역 정보가 안정적으로 표시되도록 개선했습니다.

### CODE REVIEW

File: apps/mobile/lib/features/network_map/presentation/nearby_data_source_toggle.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                  color: selected
+                      ? EasySubwayAccessibleColors.brandSignature
+                      : EasySubwayAccessibleColors.nearbyToggleIdleText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
```
[the remark this answers is not in the document]
Comment (AquilaXk) [reply apps/mobile/lib/features/network_map/presentation/nearby_data_source_toggle.dart:142]: 2aab7282에서 수정: enabled=false일 때 테두리를 제거하고 글자색을 mutedText로 낮춰 활성 상태와 시각적으로 구분되게 했다(배경은 idle fill 유지, 크기·비탭 동작 무변경). 비활성 탭 no-op·비활성 시각 상태 widget test를 함께 추가했고 flutter analyze·관련 테스트 전부 통과.

File: apps/mobile/lib/features/network_map/presentation/nearby_station_line_bar.dart [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ),
+        );
+      },
+    );
+  }
```
[the remark this answers is not in the document]
Comment (AquilaXk) [reply apps/mobile/lib/features/network_map/presentation/nearby_station_line_bar.dart:136]: 2aab7282에서 수정: 노선 바 전체를 Semantics(container) 라벨 "이전역 X, 현재역 [노선] Y, 다음역 W"(이전/다음 없으면 생략)로 감싸고 내부 개별 Text는 ExcludeSemantics로 제외했다. 시각 렌더링은 무변경이며 그룹 라벨 검증 테스트 2건을 추가했다. accessibility_baseline 포함 관련 테스트 통과.

File: apps/mobile/test/network_map_nearby_panel_test.dart [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      expect(
+        find.descendant(
+          of: find.byType(NearbyDataSourceToggle),
+          matching: find.byType(AnimatedContainer),
+        ),
+        findsNothing,
+      );
+      expect(find.byIcon(Icons.refresh), findsNothing);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (AquilaXk) [reply apps/mobile/test/network_map_nearby_panel_test.dart:309]: 2aab7282에서 수정: enabled=false에서 양쪽 세그먼트 탭이 onToggle을 호출하지 않는 검증과 비활성 시각 상태(테두리 없음·mutedText) 검증을 추가했다. 배지 폴백·그룹 라벨 테스트도 같은 커밋에 포함. 대상 테스트 파일 전부 green.

**Comment by AquilaXk** [comment]
리뷰 본문의 diff 밖 지적(배지 폴백이 역명 전체 사용) 대응: 2aab7282에서 `badgeText` 폴백을 빈 문자열로 바꾸고, 배지 텍스트가 비어 있으면 배지 원과 뒤 간격 자체를 렌더하지 않도록 조건화했다 — 노선 정보가 없는 edge case에서 빈 원이나 과축소 텍스트가 남지 않는다. 배지 렌더/미렌더 widget test 2건을 추가했고 flutter analyze·관련 테스트 전부 통과.
